.github/workflows/publish.yml

```diff
@@ -8,7 +8,102 @@ on:
 jobs:
   tests:
     uses: ./.github/workflows/test.yml
-  publish-release:
+  
+  build-installer:
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - { target: x86_64-unknown-linux-gnu, file_suffix: amd64-linux, os: ubuntu-latest}
+          - { target: x86_64-apple-darwin, file_suffix: macos, os: macos-latest}
+          - { target: x86_64-pc-windows-gnu, file_suffix: windows.exe, os: windows-latest}    
+    runs-on: ${{ matrix.job.os }}
+    defaults: 
+      run:
+        working-directory: ./micro-rdk-installer
+        shell: bash
+    steps:
+    - name: Checkout main branch code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: Set toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: 1.70.0
+        override: true
+        target: ${{ matrix.job.target }}
+        default: true
+    - name: Install x86_64 linux dependencies
+      if: ${{ matrix.job.os == 'ubuntu-latest' }}
+      run: |
+        sudo apt-get install libudev-dev pkg-config libssl-dev
+    - name: Install MacOS dependencies
+      if: ${{ matrix.job.os == 'macos-latest' }}
+      run: |
+        brew install coreutils
+        export PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
+    - name: Build installer binary
+      run: |
+        cargo build --release --target ${{ matrix.job.target }} --locked
+    - name: Rename result
+      if: ${{ matrix.job.os != 'windows-latest' }}
+      run: |
+        cp target/${{ matrix.job.target }}/release/micro-rdk-installer micro-rdk-installer-${{ matrix.job.file_suffix }}
+    - name: Rename result (Windows)
+      if: ${{ matrix.job.os == 'windows-latest' }}
+      run: |
+        cp target/${{ matrix.job.target }}/release/micro-rdk-installer.exe micro-rdk-installer-${{ matrix.job.file_suffix }}
+    - name: Check release type
+      id: check-tag
+      run: |
+        if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
+              echo "is_rc_release=true" >> $GITHUB_OUTPUT
+        else
+              echo "is_rc_release=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Upload release
+      uses: actions/upload-artifact@v2
+      with:
+        name: micro-rdk-installer-${{ matrix.job.file_suffix }}
+        path: micro-rdk-installer/micro-rdk-installer-${{ matrix.job.file_suffix }}
+
+  # github-actions does not have a runner that supports arm64 linux and cross-compilation does not seem to work for the ring crate
+  build-installer-arm64-linux:
+    runs-on: [x64, qemu-host]
```

Comment:
re runners: you're hitting an x86 box with QEMU I think? you could alternatively hit one of our arm runners or use buildjet

guessing native ARM will be faster than x86, but if you're okay with current build speed, ignore this